cranelift/docs/ir.md

```diff
@@ -142,12 +142,22 @@ Integer values have a fixed size and can be interpreted as either signed or
 unsigned. Some instructions will interpret an operand as a signed or unsigned
 number, others don't care.
 
-The support for i8 and i16 arithmetic is incomplete and use could lead to bugs.
-
 - i8
 - i16
 - i32
 - i64
+- i128
+
+Of these types, i32 and i64 are the most heavily-tested because of their use by 
+Wasmtime. There are no known bugs in i8, i16, and i128, but their use may not 
+be supported by all instructions in all backends (that is, they may cause 
+the compiler to crash during code generation with an error that an instruction
+is unsupported). 
+
+The function `valid_for_target` within the [fuzzgen function generator][fungen] 
+contains information about which instructions support which types. 
+
+[fungen]: https://github.com/elliottt/wasmtime/blob/545749b279974e87bfceada10f5f301f4503e4ce/cranelift/fuzzgen/src/function_generator.rs#L310
```

Comment:
This text is great, but I'm not sure we should use this link. At the least, I think we should link to mainline Wasmtime rather than Trevor's fork.

Also maybe we should link to whatever is on `main`, rather than a specific commit. Then we can't reliably link to the specific function, but just giving the function name is probably good enough.

So I think linking to `https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/fuzzgen/src/function_generator.rs` would be best here.